src/ol/renderer/canvas/canvasvectorlayerrenderer.js

```diff
@@ -194,6 +227,14 @@ ol.renderer.canvas.VectorLayer.prototype.prepareFrame =
 
   var extent = ol.extent.buffer(frameStateExtent,
       vectorLayerRenderBuffer * resolution);
+  var projectionExtent = viewState.projection.getExtent();
+
+  if (vectorSource.getWrapX() && viewState.projection.isGlobal() &&
```

Comment:
I did not mean to remove the `wrapX` option, I was just questioning the `global` option on `ol.proj.Projection`. But it's marked as stable as you said… So ok, statu quo.